src/core/or/policies.c

```diff
@@ -1017,6 +1017,11 @@ fascist_firewall_choose_address_ls(const smartlist_t *lspecs,
 
   tor_assert(ap);
 
+  if (lspecs == NULL || smartlist_len(lspecs) == 0) {
+    log_warn(LD_BUG, "Unknown or missing link specifiers");
```

Comment:
An empty list of link specifiers isn't a bug in this tor: they're a problem with the data that was sent to this tor.
Let's make an empty list a protocol warning.